### PR TITLE
Parse and thread through ALTER SOURCE statement

### PR DESCRIPTION
**Pull request by bkirwi**
This code parses a the new `ALTER SOURCE` statement, and threads the result through to the controller. The actual implementation is currently stubbed out.

### Motivation

Splitting this out of #14148, which implements one of the bullet points in #13125.

The parser and friends are under heavy development, including similarly-named stuff like `ALTER SYSTEM`. Landing these changes earlier than later will help to avoid some very painful merges.

### Tips for reviewer

I'm hoping that you won't need to understand how `ALTER SOURCE` is expected to work to review this PR; it's all very mechanical. If it helps, see the updated tests for how the new syntax is expected to look.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - None. (IIUC, the minor difference in error message does not count.)

### CODE REVIEW

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -3723,6 +3724,22 @@ pub fn plan_alter_secret(
...
+    _: AlterSourceStatement<Aug>,
+) -> Result<StatementDesc, PlanError> {
+    Ok(StatementDesc::new(None))
+}
+
+pub fn plan_alter_source(
+    scx: &StatementContext,
+    stmt: AlterSourceStatement<Aug>,
+) -> Result<Plan, PlanError> {
+    scx.require_unsafe_mode("ALTER SOURCE")?;
```
Comment (guswynn) [inline src/sql/src/plan/statement/ddl.rs:3738]: nice

**Review by guswynn — APPROVED** [review]
@andrioni might enjoy looking at this too, but looks reasonable to me!

`RESET` means set to the default, right?

**Review by benesch — APPROVED** [review]
LGTM too.

**Comment by bkirwi** [comment]
> RESET means set to the default, right?

Yep! _Should_ be consistent with the other `ALTER` commands we've implemented so far.

Thanks all for the review!

**Pull request merged**